pkg/controller/nebulabackup/nebula_backup_manager.go

```diff
@@ -118,65 +140,346 @@ func (bm *backupManager) Sync(backup *v1alpha1.NebulaBackup) error {
+									},
+								},
+							},
+							{
+								Name: "backup-cleanup-volume-runnable",
+								VolumeSource: corev1.VolumeSource{
+									EmptyDir: &corev1.EmptyDirVolumeSource{},
 								},
-								{
-									Name: EnvS3SecretKeyName,
-									ValueFrom: &corev1.EnvVarSource{
-										SecretKeyRef: &corev1.SecretKeySelector{
-											LocalObjectReference: corev1.LocalObjectReference{
-												Name: backup.Spec.BR.S3.SecretName,
+							},
+						},
+						RestartPolicy: corev1.RestartPolicyNever,
+						NodeSelector:  backup.Spec.NodeSelector,
+					},
+				},
+				TTLSecondsAfterFinished: aws.Int32(600),
+			},
+		}
+	} else {
+		bpCmdHead := "exec /usr/local/bin/br-ent backup full"
+		if backup.Spec.BR.BackupName != "" {
+			bpCmdHead = fmt.Sprintf("exec /usr/local/bin/br-ent backup incr --base %s", backup.Spec.BR.BackupName)
+		}
+
+		backupCmd := fmt.Sprintf("%s --meta %s --storage s3://%s --s3.access_key $%s --s3.secret_key $%s --s3.region %s --s3.endpoint %s",
+			bpCmdHead, metaAddr, backup.Spec.BR.S3.Bucket, EnvS3AccessKeyName, EnvS3SecretKeyName, backup.Spec.BR.S3.Region, backup.Spec.BR.S3.Endpoint)
+
+		return &batchv1.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      backup.Name,
+				Namespace: backup.Namespace,
+			},
+			Spec: batchv1.JobSpec{
+				Parallelism:  pointer.Int32(1),
+				Completions:  pointer.Int32(1),
+				BackoffLimit: pointer.Int32(0),
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						ImagePullSecrets: backup.Spec.ImagePullSecrets,
+						Containers: []corev1.Container{
+							{
+								Name:            "backup",
+								Image:           backup.Spec.ToolImage,
+								ImagePullPolicy: backup.Spec.ImagePullPolicy,
+								Env: []corev1.EnvVar{
+									{
+										Name: EnvS3AccessKeyName,
+										ValueFrom: &corev1.EnvVarSource{
+											SecretKeyRef: &corev1.SecretKeySelector{
+												LocalObjectReference: corev1.LocalObjectReference{
+													Name: backup.Spec.BR.S3.SecretName,
+												},
+												Key: br.S3AccessKey,
+											},
+										},
+									},
+									{
+										Name: EnvS3SecretKeyName,
+										ValueFrom: &corev1.EnvVarSource{
+											SecretKeyRef: &corev1.SecretKeySelector{
+												LocalObjectReference: corev1.LocalObjectReference{
+													Name: backup.Spec.BR.S3.SecretName,
+												},
+												Key: br.S3SecretKey,
 											},
-											Key: br.S3SecretKey,
 										},
 									},
 								},
+								Command: []string{"/bin/sh", "-ecx"},
+								Args:    []string{backupCmd},
 							},
-							Command: []string{"/bin/sh", "-ecx"},
-							Args:    []string{backupCmd},
 						},
+						RestartPolicy: corev1.RestartPolicyNever,
+						NodeSelector:  backup.Spec.NodeSelector,
 					},
-					RestartPolicy: corev1.RestartPolicyNever,
-					NodeSelector:  backup.Spec.NodeSelector,
 				},
+				TTLSecondsAfterFinished: aws.Int32(600),
```

Comment:
invoke k8s.io/utils/pointer

Reply:
@MegaByte875 Got it, will use k8s.io/utils/pointer instead.

Reply:
Fixed. Used k8s.io/utils/pointer in the end since it was used elsewhere in the code.